source/MaterialXGenShader/Syntax.h

```diff
@@ -171,6 +171,16 @@ class Syntax
     /// on the name string if there is a name collision.
     virtual string getVariableName(const string& name, const TypeDesc* type, IdentifierMap& identifiers) const;
 
+    /// Given an input specification attempt to remap this to an enumeration which is accepted by
+    /// the shader generator. The enumeration may be converted to a different type than the input.
+    /// @param value The value string to remap.
+    /// @param type The type of the value to remap,
+    /// @param enumNames Type enumeration names
+    /// @param result Enumeration type and value (returned).
+    /// @return Return true if the remapping was successful.
+    virtual bool remapEnumeration(const string& value, const TypeDesc* type, const string& enumNames,
```

Comment:
Seems to make more sense to have enum remapping part of syntax so it's self-contained.

Reply:
Yep, looks good!